include/ps/kv_app.h

```diff
@@ -425,6 +431,26 @@ struct KVServerDefaultHandle {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+template <typename Val>
+void KVServer<Val>::RegisterRecvBuffer(int worker_id, SArray<Key>& keys,
+                                       const SArray<Val>& vals,
+                                       const SArray<int>& lens,
+                                       int cmd) {
+  Message msg;
+  msg.meta.request = true;
+  msg.meta.push = true;
+  msg.meta.head = cmd;
+  msg.meta.sender = worker_id;
+  if (keys.size()) {
+    msg.AddData(keys);
+    msg.AddData(vals);
+    if (lens.size()) {
+      msg.AddData(lens);
+    }
+  }
+  Postoffice::Get()->van()->RegisterRecvBuffer(msg);
```

Comment:
so, inside `van()->RegisterRecvBuffer` need to take key from `msg.data[0]`, right?